Experiment/salmon

slabasan opened the pull request:
Depends on #703 to pass lint CI checks.

~~Blocked by:~~
- [x] ~~#623 -- debug errors when updating spack version~~ Remove `depends_on()` and no longer need a newer spack
- [x] ~~Babelstream package in benchpark vs updated in spack~~ Fixed in develop

## Description

- [x] Add salmon experiment
- [x] Replaces #391, moves branch to upstream so CI tests run

## Adding/modifying a benchmark (docs: [Adding a Benchmark](https://software.llnl.gov/benchpark/add-a-benchmark.html))

- [x] If modifying the source code of a benchmark: create, self-assign, and link here a follow up issue with a link to the PR in the benchmark repo.
- [x] If package.py upstreamed to Spack is insufficient, add/modify `repo/benchmark_name/package.py` plus: create, self-assign, and link here a follow up issue with a link to the PR in the Spack repo.
- [x] If application.py upstreamed to Ramble is insufficient, add/modify `repo/benchmark_name/application.py` plus: create, self-assign, and link here a follow up issue with a link to the PR in the Ramble repo.
- [x] Tags in Ramble's `application.py` or in `repo/benchmark_name/application.py` will appear in the [docs catalogue](https://software.llnl.gov/benchpark/benchmark-list.html)
- [x] Add/modify an `experiments/benchmark_name/experiment.py` to define a single node and multi-node experiments
- [x] Add/modify a dry run unit test in `.github/workflows/run.yml`

dyokelson committed:
change to default-compiler, try out on dane tomorrow

dyokelson committed:
Add CI on x86 and default-compiler to cmake

dyokelson committed:
remove .tpl file

slabasan committed:
running on ruby

use a consistent compiler between ruby and fugaku

slabasan committed:
add salmon application.py

slabasan committed:
add dynamic dry runs on ruby and fugaku

slabasan committed:
rename salmon directory, class and format

slabasan committed:
salmon: add openmp variant

slabasan committed:
add local salmon package and patches

remove variant from application.py file

slabasan committed:
fix spack package name for salmon

slabasan committed:
salmon package.py and experiment.py fixes

- openmp is not a package in spack, so cannot depend_on()
- add fallow-argument-mismatch flag for gcc10 and later

pearce8 commented on checkout-versions.yaml:
```diff
@@ -5,4 +5,4 @@
 versions:
   ramble: f5970aa     # develop on 1/08/2025 (newer than 0.5.0 release)
-  spack: 0b3b49b      # develop-2024-06-28 (newer than 0.22.0 release)
+  spack: c6d4037      # v0.23.0 (2024-11-13)
```
I pulled the spack update version into its own PR https://github.com/LLNL/benchpark/pull/623; lets test it with other benchmarks and merge that PR first.

slabasan replied:
Yes you beat me to it. Needed a new spack version because it was looking for `Package 'c' not found.` when trying to install salmon.

pearce8 commented:
@slabasan What was the reason to pull in a new version of Spack here?  I think it is causing some of the issues with your tests, so I pulled out the version update into its own PR #623 to try to figure that part out.

slabasan commented on repo/salmon-tddft/package.py:
```diff
@@ -0,0 +1,110 @@
+    version("2.0.0", sha256="c3bb80bc5d338cba21cd8f345acbf2f2d81ef75af069a0a0ddbdc0acf358456c")
+    version("1.2.1", sha256="a5045149e49abe9dd9edefe00cd1508a1323081bc3d034632176b728effdbaeb")
+
+    depends_on("c", type="build")  # generated
```
Remove these lines, then don't need a newer spack

slabasan committed:
rebase and update

remove depends_on calls in salmon's package.py to avoid needing newer
spack version

michaelmckinsey1 commented on experiments/salmon-tddft/experiment.py:
```diff
@@ -0,0 +1,55 @@
+        # get package version
+        app_version = self.spec.variants["version"][0]
+
+        system_specs = {}
+        system_specs["compiler"] = "default-compiler"
+
+        self.add_spack_spec(
+            self.name, [f"salmon-tddft@{app_version}", system_specs["compiler"]]
+        )
```
```suggestion
        self.add_package_spec(
            self.name, [f"salmon-tddft@{app_version}"]
        )
```

slabasan replied:
Can you help me understand why we need to remove more of this code?

michaelmckinsey1 replied:
https://github.com/LLNL/benchpark/pull/655 Removed experiment handles. 655 was merged after this PR was opened so you would not have noticed changes to the other experiments until the recent rebase.

The only part we need now is the package spec

```
self.add_package_spec(
   self.name, [f"salmon-tddft@{app_version}"]
)

slabasan commented:
The lint CI check caught a mis-spelling of optimization in `repo/salmon_tddft/fjmpi.patch`. But is also thinking that `nd` is a mis-spelled `2nd`. This is a false positive error. Should lint CI be checking these files?

slabasan committed:
codespell: ignore patch files

slabasan committed:
move salmon tests to end of dryrunexperiments section in run.yml

slabasan committed:
add experiment setup

slabasan committed:
remove second call to create salmon/openmp dir

slabasan committed:
remove duplicate fugaku system init

slabasan committed:
removing experiment handles, change already merged into develop

michaelmckinsey1 reviewed (dismissed):
Looks good once lint passing

slabasan committed:
use double asterisk wildcard

Co-authored-by: Alec Scott <scott112@llnl.gov>

slabasan commented:
This one is (finally) ready!

dyokelson reviewed (commented):
Do we want all the tests in the run.yml or just in the new format? Just not sure if lines 265-285 are still needed or it would be best to just keep the later versions.

We should have the Riken folks test this out on Fugaku as well.

michaelmckinsey1 committed:
black
